### PR TITLE
Add support for silently filtered words

### LINKED ISSUE #39: Support silent filters
We want to be able to detect when certain keywords are mentioned and notify moderators without removing the message.

This requires an extension to the FilterWord model with an optional `silent_filter` flag, as well as an optional flag in the `/filter add` command or a way to update an existing word to toggle the flag on. 

We also need to update places using the filter word to not delete the message when this is set to true.

### PR DESCRIPTION
**Pull request by SlimShadyIAm**
Closes #39 

## Description
We want to be able to detect when certain keywords are mentioned and notify moderators without removing the message.

## Changes
- Extend `FilterWord` model with new `silent_filter` property which defaults to false
- New command `/silent_filter` to toggle silent filter mode on already filtered words
- `/filter list` shows when a word is silently filtered
- Ignore silently filtered words in commands that typically stop executing/return when a filtered word is triggered (i.e when you search a package in Canister with a filtered word as the query)
- Generate report for silently filtered words, but don't delete the word, ratelimit the user or send them a message privately

### CODE REVIEW

File: config.example.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "birthday": 123,
+    "member_ultra": 123,
+    "member_one": 123,
+    "member_edition": 123,
+    "member_pro": 123,
+    "member_plus": 123,
+    "sub_news": 123,
+    "aaron_role": 123
+  }
+}
```
Comment (BBaoVanC) [inline config.example.json]: nit: missing newline at end of file

File: data/model/filterword.py [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import mongoengine
 
 class FilterWord(mongoengine.EmbeddedDocument):
-    # _id    = mongoengine.ObjectIdField(required=True, default=mongoengine.ObjectId., unique=True, primary_key=True)
     notify               = mongoengine.BooleanField(required=True)
+    silent_filter         = mongoengine.BooleanField(default=False)
```
Comment (BBaoVanC) [inline data/model/filterword.py]: nit: indentation misaligned; single extra space before `=`

File: utils/framework/filter.py [hunk 3]
```diff
@@ -49,6 +49,13 @@ async def find_triggered_filters(input, member: discord.Member) -> List[FilterWo
             words_found.append(word)
     return words_found
 
+def ignorable_words(triggered_filter_words: List[FilterWord]):
```
Comment (BBaoVanC) [inline utils/framework/filter.py]: `ignorable_words` doesn't sound like a very descriptive name, maybe `has_only_silent_words`

Or if you want, you can flip the logic (return true if at least one word is not silently filtered) and call it something different like `has_notifiable_word`

**Review by BBaoVanC — COMMENTED** [review]
I am not familiar enough with the codebase to do a meaningful review, so that is what I have found

File: utils/framework/filter.py (hunk 3 above)
Comment (SlimShadyIAm) [reply utils/framework/filter.py]: Thanks, I've renamed it to `has_only_silent_filtered_words`

**Pull request merged**